### PR TITLE
RHDEVDOCS 4233 add information on Pipelines authentication using workspaces, container registries

### PR DESCRIPTION
**Pull request by mramendi**
Version(s):

cp to pipelines-docs-1.10, pipelines-docs-1.11, pipelines-docs-1.12, pipelines-docs-1.13, pipelines-docs-1.14

Issue:

RHDEVDOCS 4233

Link to docs preview:

https://70384--docspreview.netlify.app/openshift-pipelines/latest/secure/authenticating-pipelines-repos-using-secrets

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/op-using-ssh-authentication-in-git-type-tasks.adoc
Comment (vdemeester) [inline modules/op-using-ssh-authentication-in-git-type-tasks.adoc:24]: `ln -s /tekton/home/.ssh …` or better `ln -s $HOME/.ssh …`.

**Review by vdemeester — COMMENTED** [review]
LGTM, one nit.

**Review by vdemeester — COMMENTED** [review]
I also agree that the `modules/op-limiting-secret-access-to-specific-steps.adoc` could be removed.

**Review by vdemeester — APPROVED** [review]

**Review by ppitonak — APPROVED** [review]

File: modules/op-using-secrets-as-a-nonroot-user.adoc
Comment (skopacz1) [inline modules/op-using-secrets-as-a-nonroot-user.adoc:18]: I know that this already existed when you made this change, but per our [contribution guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-to-external-websites) we "cannot link to a repository that is hosted on www.github.com". "pipeline" is a repository for the tektoncd organization, right? Do you happen to know if this link was included by some special exception or not? If not, I feel like this should ideally be removed, as unfortunate as it would be to get rid of this seemingly useful example.

File: modules/op-using-ssh-authentication-in-git-type-tasks.adoc
Comment (skopacz1) [inline modules/op-using-ssh-authentication-in-git-type-tasks.adoc:7]: Tiny adverb position and [active voice](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#voice) nit:
```suggestion
You can directly invoke Git commands in the steps of a task and use SSH authentication, but you must perform an additional step.
```

File: modules/op-using-ssh-authentication-in-git-type-tasks.adoc
Comment (skopacz1) [inline modules/op-using-ssh-authentication-in-git-type-tasks.adoc:30]: This is even further out of scope for this review, so I'm only pointing it out this time, but this is another instance of a problematic link to a GitHub-hosted repository.

File: secure/authenticating-pipelines-repos-using-secrets.adoc
Comment (skopacz1) [inline secure/authenticating-pipelines-repos-using-secrets.adoc:22]: I think since all the procedural modules in this section use the oc CLI, it would be good to have a Prereqs section defining that as a prereq:
```suggestion

[id="providing-secrets-using-service-accounts-prereqs_{context}"]
=== Prerequisites
* You have installed the OpenShift CLI (`oc`)

```

Not honestly sure if the id/anchor is needed. Optionally, if you prefer then you can add this as a prereq in each procedural module that includes the use of `oc` instead of putting it here.

File: modules/op-configuring-registry-authentication-sa.adoc
Comment (skopacz1) [inline modules/op-configuring-registry-authentication-sa.adoc:12]: I'd add a prereq here for the oc CLI:
```suggestion

.Prerequisites
* You have installed the OpenShift CLI (`oc`)

```

**Review by skopacz1 — COMMENTED** [review]
Left comments mostly related to style guide and formatting nuances, but this is seriously nice work, great job! Let me know if you have any questions about my feedback.

I am still (relatively) new to reviewing, so please wait until @adellape approves of my review before implementing any feedback. Thanks!

File: modules/op-using-secrets-as-a-nonroot-user.adoc
Comment (mramendi) [reply modules/op-using-secrets-as-a-nonroot-user.adoc:18]: I was under the impression that repositories maintained by Red Hat engineers were excepted, will await @adellape comment

File: modules/op-configuring-container-registry-authentication-using-workspaces.adoc
Comment (mramendi) [reply modules/op-configuring-container-registry-authentication-using-workspaces.adoc:4]: I was under the impression that this was no longer a requirement but I can be mistaken, will await @adellape comment

File: modules/op-configuring-basic-authentication-for-git.adoc
Comment (mramendi) [reply modules/op-configuring-basic-authentication-for-git.adoc:60]: I disagree regarding numbered steps, I think at last one ISG example has unnumbered ones - the one with "Click Configure to do any of the following tasks"

File: _topic_maps/_topic_map.yml
Comment (mramendi) [reply _topic_maps/_topic_map.yml:110]: I considered this but I kinda think the addition of containers is too important

File: modules/op-configuring-container-registry-authentication-using-workspaces.adoc
Comment (mramendi) [reply modules/op-configuring-container-registry-authentication-using-workspaces.adoc:42]: I made a shot at this suggestion. Not sure if this works better, though.

File: modules/op-configuring-container-registry-authentication-using-workspaces.adoc
Comment (mramendi) [reply modules/op-configuring-container-registry-authentication-using-workspaces.adoc:18]: Your suggestion led to inconsistency with the precious line and callout, so I made it fully consistent instead

File: modules/op-configuring-git-ssh-authentication-using-workspaces.adoc
Comment (mramendi) [reply modules/op-configuring-git-ssh-authentication-using-workspaces.adoc:19]: Again. made all lines fully consistent instead

File: modules/op-configuring-basic-authentication-for-git.adoc
Comment (mramendi) [reply modules/op-configuring-basic-authentication-for-git.adoc:60]: I also think that using the imperative would not be consistent with "Use one of the following examples:" as the introduction (which you also suggested and I do agree). However there was no second-level offset, I fixed that now.

File: modules/op-types-annotation-secrets.adoc
Comment (mramendi) [reply modules/op-types-annotation-secrets.adoc:92]: Unfortunately this change would make it inconsistent with previous examples in the file

File: modules/op-using-ssh-authentication-in-git-type-tasks.adoc
Comment (mramendi) [reply modules/op-using-ssh-authentication-in-git-type-tasks.adoc:30]: Again, I believe the repo is maintained by Red Hat engineers. There are many links to this and other similar repos in this doc set and removing them would be a major problem. As far as I am aware, the intend of the guidance is to disallow repos that we don't control and that therefore can be volatile.

File: secure/authenticating-pipelines-repos-using-secrets.adoc
Comment (mramendi) [reply secure/authenticating-pipelines-repos-using-secrets.adoc:22]: Did this before the subdivisions.

**Comment by mramendi** [comment]
@skopacz1 I made the changes I was reasonably sure I had an idea about and will now await @adellape review.

Note: I am aware of the need to squash the commits before requesting the merge review. I did not yet squash the commits because there might be more change requests in this peer review.

File: modules/op-using-secrets-as-a-nonroot-user.adoc
Comment (adellape) [reply modules/op-using-secrets-as-a-nonroot-user.adoc:18]: My understanding was https://github.com/redhat-developer was the only exception.

But yes, otherwise this reference should either be removed or ported and QE-verified to this repo in some way.

File: modules/op-using-ssh-authentication-in-git-type-tasks.adoc
Comment (adellape) [reply modules/op-using-ssh-authentication-in-git-type-tasks.adoc:30]: The examples could still change out from under us and not necessarily have been QE verified, particularly for the docs version they might be appearing in (especially if just targeting the `main` branch like this example is).

**Review by adellape — COMMENTED** [review]
Added a few more comments and replies, but peer review LGTM 👍

File: modules/op-using-ssh-authentication-in-git-type-tasks.adoc
Comment (mramendi) [reply modules/op-using-ssh-authentication-in-git-type-tasks.adoc:30]: I have requested Pipelines QE feedback on these examples.

File: modules/op-using-secrets-as-a-nonroot-user.adoc
Comment (mramendi) [reply modules/op-using-secrets-as-a-nonroot-user.adoc:18]: I have requested Pipelines QE feedback on these examples.

**Comment by mramendi** [comment]
@adellape thanks! I have made the requested changes. I have asked our QE regarding the tektoncd github repo.

**Comment by aireilly** [comment]
This branch has been repointed to target pipelines-docs-main. pipelines-docs is no longer in use.

File: modules/op-using-secrets-as-a-nonroot-user.adoc
Comment (mramendi) [reply modules/op-using-secrets-as-a-nonroot-user.adoc:18]: This one issue kept the PR dormant for two months. But it is now resolved, I have the new repo.

**Comment by mramendi** [comment]
This was dormant for two months after a completed peer review, because the peer review pointed out a link to an upstream repository and getting an RH repository to replace it took time. This is done now. Other peer review notes were fixed back then, and there were no further changes. Requesting merge.

File: secure/authenticating-pipelines-repos-using-secrets.adoc
Comment (sheriff-rh) [inline secure/authenticating-pipelines-repos-using-secrets.adoc:36]: No anchor ID on this H3?

File: about/understanding-openshift-pipelines.adoc
Comment (sheriff-rh) [inline about/understanding-openshift-pipelines.adoc:40]: Needs an anchor ID. https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: create/creating-applications-with-cicd-pipelines.adoc
Comment (sheriff-rh) [inline create/creating-applications-with-cicd-pipelines.adoc:59]: Needs an anchor ID. https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

**Review by sheriff-rh — CHANGES_REQUESTED** [review]
I noticed a few missing anchor IDs, which may not seem like a big deal but this is the kind of thing that if it doesn't get addressed now, will become a problem on the customer portal publication - even if it passes a github test build.

**Comment by mramendi** [comment]
@sheriff-rh I have added the anchor IDs. For the additional resources, I added them to every additional resources section in the affected assemblies in order to ensure consistency.

**Comment by mramendi** [comment]
I have made the requested change but am unable to reach @sheriff-rh , I hope someone can merge this

**Review by sheriff-rh — APPROVED** [review]
Changes LGTM, merging.

**Comment by sheriff-rh** [comment]
Looks like a merge conflict on 1.10 @mramendi - I was able to merge 1.11-1.14 cleanly though :+1:

**Comment by mramendi** [comment]
Thanks! Let's forget 1.10

**Comment by mramendi** [comment]
Pipelines 1.10 is no longer supported, so I will not adapt the change to Pipelines 1.10 docs, pending their possible removal or marking as unsupported.

**Pull request merged**
